src/protocol/sort/bit_permutation.rs

```diff
@@ -16,13 +21,17 @@ use futures::future::try_join_all;
 /// by K. Chida, K. Hamada, D. Ikarashi, R. Kikuchi, N. Kiribuchi, and B. Pinkas
 /// <https://eprint.iacr.org/2019/695.pdf>.
 #[derive(Debug)]
-pub struct BitPermutation<'a, F> {
-    input: &'a [Replicated<F>],
+pub struct BitPermutation<'a, F, S> {
+    input: &'a [S],
+    _marker: PhantomData<F>,
 }
```

Comment:
First of all, if it helps at all, feel free to eliminate the `new` function and just turn this protocol into a single public async function as we already have done with many other protocols.

Secondly, what is this `PhantomData<F>`? I have not used this before. I don't understand why this is necessary.

Thirdly, I'm surprised that the input doesn't explain that it's connected to the field `F` in any way. It's a secret sharing of a field element, so I was expecting to see something like: `&'a [S<F>]`. Maybe if this was just a function argument to a public async function and not a property in a struct this would be easier?

Reply:
Thanks for pointing this out, making the change